pkg/segment/structs/evaluationstructs.go

```diff
@@ -1239,6 +1239,23 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 			return math.Abs(left), nil
 		case "ceil":
 			return math.Ceil(left), nil
+		case "log":
+			if left < 0 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: Negative values cannot be used for logarithm: %v", left)
+			}
+			if right != 2 && right != 10 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: Logarithm base must be either 2 or 10: %v", right)
+			}
```

Comment:
Is this in the splunk documentation? I'm not finding it

Reply:
yes, it's here https://docs.splunk.com/Documentation/SCS/current/SearchReference/MathematicalFunctions#log.28.26lt.3Bnum.26gt.3B.2C.26lt.3Bbase.26gt.3B.29

Reply:
Those are just examples where `2` and `10` are valid values, but I don't see anything indicating all other values are invalid